compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -446,29 +448,32 @@ void addMultiTilingExpertPassPipeline(
   // Apply tile and fuse to all the non-distribution fusable levels. Skip
   // distribution level as that level has been fused already.
   if (allFusableLevels.size() > 1) {
-    ArrayRef<int64_t> nonDistFusableLevels(allFusableLevels.begin() + 1,
-                                           allFusableLevels.end());
-    for (int64_t level : nonDistFusableLevels) {
-      nestedModulePM.addNestedPass<func::FuncOp>(
-          createLLVMCPUTileAndFusePass(level));
-      nestedModulePM.addNestedPass<func::FuncOp>(
-          createFuseTensorPadWithConsumerPass());
-      nestedModulePM.addNestedPass<func::FuncOp>(
-          createConcretizePadResultShapePass());
-    }
-  }
+    llvm::SmallSetVector<int64_t, 4> fusableLevels(allFusableLevels.begin(),
+                                                   allFusableLevels.end());
+    for (int i = 0; i < tilingConfig.getNumTilingLevels(); ++i) {
+      if (i == tilingConfig.getDistributionLevel()) continue;
+      if (fusableLevels.contains(i)) {
```

Comment:
Not sure I understand the copy in 451. We should also take advantage of the fact that fusable levels are contiguous and avoid lookups (what the previous code does).